src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -468,6 +468,10 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
                 {
                     await Authenticate(username, timeout, async, cancellationToken);
 
+#if DEBUG
+                    ReadBuffer.EnableFrameTracking = true;
```

Comment:
Any particular reason this shouldn't always be on (in DEBUG)? We could also use it to catch bugs in the Startup message and in authentication...